src/Aspire.Hosting.Azure/SqlServer/AzureSqlDatabaseResourceBuilderExtensions.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Azure.Provisioning.Sql;
+
+namespace Aspire.Hosting.Azure;
+
+/// <summary>
+/// Contains extension methods for enabling Aspire's Redis resource to be deployed to Azure.
+/// </summary>
+public static class AzureSqlDatabaseResourceBuilderExtensions
+{
+    /// <summary>
+    /// TODO:
+    /// </summary>
+    /// <param name="builder">TODO:</param>
+    /// <param name="configureResource">TODO:</param>
+    /// <returns>TODO:</returns>
+    public static IResourceBuilder<SqlServerServerResource> PublishAsAzureSqlDatabase(this IResourceBuilder<SqlServerServerResource> builder, Action<BicepGenerationContext>? configureResource = null)
+    {
+        return builder.WithManifestPublishingCallback(context => ProvisioningExtensions.WriteBicepResourceToManifest(context, builder.Resource))
+                      .WithBicepGenerationCallback(configureResource ?? GenerateDefaultSqlServerResource);
+    }
+
+    private static void GenerateDefaultSqlServerResource(BicepGenerationContext context)
+    {
+        var sql1 = new SqlServer(context, context.Resource.Name + "1");
+        var sql2 = new SqlServer(context, context.Resource.Name + "2");
```

Comment:
This surplus SqlServer provisioning resource is here to illustrate a problem. Currently in Azure Provisioning it doesn't seem to be possible to add two resources of the same type to the same scope where a child resource is added (the API surface doesn't seem to allow to specify which parent to use ... it blows up).

Reply:
I would need to see the implementation here I added this unit test into our cdk and it works ifne

```C#
        [Test]
        public void DoubleSqlServer()
        {
            var infra = new TestInfrastructure();
            SqlServer sqlServer1 = new SqlServer(infra, "sqlserver1");
            SqlServer sqlServer2 = new SqlServer(infra, "sqlserver2");
            infra.Build(GetOutputPath());
        }
```

Resulting bicep
```bicep
targetScope = subscription


resource resourceGroup_I6QNkoPsb 'Microsoft.Resources/resourceGroups@2023-07-01' = {
  name: 'rg-TEST'
  location: 'westus'
  tags: {
    azd-env-name: 'TEST'
  }
}

resource sqlServer_tXc1185GI 'Microsoft.Sql/servers@2022-08-01-preview' = {
  scope: resourceGroup_I6QNkoPsb
  name: 'sqlserver1-TEST'
  location: 'westus'
  properties: {
    administratorLogin: 'sqladmin'
    administratorLoginPassword: '00000000-0000-0000-0000-000000000000'
    version: '12.0'
    minimalTlsVersion: '1.2'
    publicNetworkAccess: 'Enabled'
  }
}

resource sqlServer_QdKPtcbIX 'Microsoft.Sql/servers@2022-08-01-preview' = {
  scope: resourceGroup_I6QNkoPsb
  name: 'sqlserver2-TEST'
  location: 'westus'
  properties: {
    administratorLogin: 'sqladmin'
    administratorLoginPassword: '00000000-0000-0000-0000-000000000000'
    version: '12.0'
    minimalTlsVersion: '1.2'
    publicNetworkAccess: 'Enabled'
  }
}

```

Reply:
It throws when you add a database

Reply:
FYI there is a new alpha 1.0.0-alpha.20240220.1, that allows you to specify a sqlServer for the database being added.

Reply:
The error you were getting was because it was ambiguous which SqlServer to add the SqlDatabase too.  The ctor for all resources should have an optional parent parameter to specify if you have multiple.

I am imagining from Aspire code perspective we would have something like
```c#
builder.AddSqlServer("myserver")
       .AddSqlDatabase("mydb");
```
With this you will know the parent given the call chain and can pass this information into cdk.  However if you do this

```c#
builder.AddSqlServer("myserver");
builder.AddSqlDatabase("mydb");
```
In the second line the database will try to find its parent in its current scope and attach itself unless there are more than one at which point it will error asking you to specify.